packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart

```diff
@@ -15,9 +23,270 @@ class DiffPane extends StatefulWidget {
+    snapshots.removeRange(index.value, index.value + 1);
+    index.value = index.value - 1;
+  }
+
+  @override
+  void dispose() {
+    for (var e in snapshots) {
+      e.dispose();
+    }
+    super.dispose();
+  }
+}
+
 class _DiffPaneState extends State<DiffPane> {
+  final controller = _DiffPaneController();
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: controller,
+      builder: (_, __) => Split(
+        axis: Axis.horizontal,
+        initialFractions: const [0.2, 0.8],
+        minSizes: const [80, 80],
+        children: [
+          Column(
+            children: [
+              _ListControlPane(controller: controller),
+              Expanded(
+                child: _SnapshotList(controller: controller),
+              ),
+            ],
+          ),
+          Column(
+            children: [
+              if (controller.selected is SnapshotListItem)
+                _ContentControlPane(controller: controller),
+              Expanded(
+                child: _SnapshotListContent(item: controller.selected),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+}
+
+class _SnapshotList extends StatelessWidget {
+  const _SnapshotList({Key? key, required this.controller}) : super(key: key);
+  final _DiffPaneController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView.builder(
+      controller: controller.scrollController,
+      shrinkWrap: true,
+      itemCount: controller.snapshots.length,
+      itemBuilder: (context, index) {
+        return ListTile(
+          title: _SnapshotListTitle(
+            item: controller.snapshots[index],
+            selected: index == controller.index.value,
+          ),
+          onTap: () => controller.index.value = index,
+          key: GlobalObjectKey(controller.snapshots[index]),
+        );
+      },
+    );
+  }
+}
+
+class _SnapshotListTitle extends StatelessWidget {
+  const _SnapshotListTitle({
+    Key? key,
+    required this.item,
+    required this.selected,
+  }) : super(key: key);
+  final DiffListItem item;
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: item,
+      builder: (context, child) => Row(
+        children: [
+          _SelectionBox(selected: selected),
+          const SizedBox(width: denseRowSpacing),
+          if (item.isProcessing) const _ProgressIndicator(),
+          if (item.isProcessing) const SizedBox(width: denseRowSpacing),
+          Expanded(
+            child: Text(item.name, overflow: TextOverflow.ellipsis),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ProgressIndicator extends StatelessWidget {
+  const _ProgressIndicator({Key? key}) : super(key: key);
+  static const _progressIndicatorSize = 16.0;
```

Comment:
does this need to be scaled for screen density?

Reply:
May be. I did not do this before. Can you point me to example, please?

Reply:
I've notices this density being used in certain places
https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/inspector/inspector_breadcrumbs.dart#L36
I've also seen it used in conjunction with scaleByFontFactor
https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/inspector/inspector_tree.dart

I'm not really sure when or how to use these tbh but since a progress indicator might be something we want to be well proportioned, I figured it would be worth bringing it up.
Perhaps @kenzieschmoll could school us?

Reply:
We use `scaleByFontFactor` to make DevTools respect the IDE / browser font size. We should use it here if we want it to scale. Here is an example of where we use `smallProgressSize`, which is a scaled value.